guides/search.md

```diff
@@ -0,0 +1,289 @@
+
+```python
+from opensearchpy import OpenSearch
+# Create an OpenSearch client with appropriate hosts and connection details
+client = OpenSearch(hosts=['localhost'])
+# Define the search query with scroll and pagination options
+search_body = {
+    "query": {
+        "match": {
+            "title": "dark knight"
+        }
+    },
+    "scroll": "1m",  # Set the scroll duration to 1 minute
+    "size": 2
+}
+# Perform the initial search operation on the 'movies' index with the defined query and scroll options
+page_1 = client.search(
+    index='movies',
+    body=search_body
+)
+# Extract the scroll_id from the response
+scroll_id = page_1['_scroll_id']
+# Perform the scroll operation to get the next page of results
+page_2 = client.scroll(
+    scroll_id=scroll_id,
+    scroll='1m'
+)
+# Extract the scroll_id from the response
+scroll_id = page_2['_scroll_id']
+# Perform another scroll operation to get the third page of results
+page_3 = client.scroll(
+    scroll_id=scroll_id,
+    scroll='1m'
+)
+# Extract the hits from each page of results
+hits_page_1 = page_1['hits']['hits']
+hits_page_2 = page_2['hits']['hits']
+hits_page_3 = page_3['hits']['hits']
+```
+
+
+
+### Pagination with Point in Time
+
+The scroll example above has one weakness: if the index is updated while you are scrolling through the results, they will be paginated inconsistently. To avoid this, you should use the "Point in Time" feature. The following example demonstrates how to use the `point_in_time` and `pit_id` parameters to paginate through the search results:
+
+```python
+from opensearchpy import OpenSearch
+# Create an OpenSearch client with appropriate hosts and connection details
+client = OpenSearch(hosts=['localhost'])
+# Define the search query with sorting and pagination options
+search_body = {
+    "query": {
+        "match": {
+            "title": "dark knight"
+        }
+    },
+    "sort": [
+        {
+            "year": {
+                "order": "asc"
+            }
+        }
+    ]
+}
+# Create a point in time
+pit = client.create_pit(index='movies', keep_alive='1m')
+# Include pit info in the search body
+pit_search_body = search_body.copy()
+pit_search_body['pit'] = {
+    'id': pit['pit_id'],
+    'keep_alive': '1m'
+}
+# Get the first 3 pages of results
+page_1 = client.search(size=2, body=pit_search_body)['hits']['hits']
+page_2 = client.search(
+    size=2,
+    body=pit_search_body.copy().update({'search_after': page_1[-1]['sort']})
+)['hits']['hits']
+
+page_3 = client.search(
+    size=2,
+    body=pit_search_body.copy().update({'search_after': page_2[-1]['sort']})
+)['hits']['hits']
+# Print out the titles of the first 3 pages of results
+print([hit['_source']['title'] for hit in page_1])
+print([hit['_source']['title'] for hit in page_2])
+print([hit['_source']['title'] for hit in page_3])
+# Delete the point in time
+client.delete_pit(body={'pit_id': pit['pit_id']})
```

Comment:
The API is `delete_point_in_time`. See https://github.com/opensearch-project/opensearch-py/blob/main/USER_GUIDE.md#point-in-time-api